### PR TITLE
CreateOrder Extension point example with unit tests

### PR DESCRIPTION
**Pull request by szelenin-salesforce**
An example for new [Checkout Create Order extension point](https://developer.salesforce.com/docs/commerce/salesforce-commerce/guide/CheckoutCreateOrder.html) with unit tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,40 +2,45 @@
 
 This repository contains a reference implementation of the Commerce Extensibility Framework. It includes sample code for extending the following services:
 
-- Pricing
-- Shipping Calculator
-- Tax Calculator
-- Tax Service
+- [Domain Extensions for Cart Calculate API](#domain-extensions-for-cart-calculate-api):
```
Comment (szelenin-salesforce) [inline README.md:5]: Reorganized readme a little bit so it reflects the same structure as we have in the docs

File: commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleIntegrationTest.cls [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        // Act
+        Test.startTest();
+        CartExtension.CreateOrderResponse response = createOrder.createOrder(new CartExtension.CreateOrderRequest(cart, '123'));
+        Test.stopTest();
+        // Assert
+        Order order = response.getOrderGraph().getOrder();
+        Assert.areEqual('Total rounded up cost of products: ' + (2.0 + 3.0), order.Description);
+    }
+
+        private static CartExtension.Cart arrangeCart(Decimal[] itemsSalePrice) {
```
Comment (yan-huang-sf) [inline commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleIntegrationTest.cls:33]: Indentation does not look right for this function

File: commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            CartExtension.Cart cart = request.getCart();
+            CartExtension.OrderGraph orderGraph = new CartExtension.OrderGraph();
+            Order order = orderGraph.getOrder();
+            System.assert(order.Id != null);
+    
+            order.AccountId = cart.getAccountId();
+            order.OwnerId = cart.getOwnerId();
+            order.SalesStoreId = cart.getWebStoreId();
+            // cart does not support PoNumber yet
+            //order.PoNumber = cart.getPoNumber();
```
Comment (yan-huang-sf) [inline commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls:49]: Do we want to keep the commented out code?

Comment (yan-huang-sf) [reply commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls:49]: As we going to use HSOI create order, we need to ensure those fields in the Order entities are available in summary entities.

File: commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleIntegrationTest.cls (hunk 2 above)
Comment (szelenin-salesforce) [reply commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleIntegrationTest.cls:33]: fixed

File: commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls (hunk 3 above)
Comment (szelenin-salesforce) [reply commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls:49]: to be on the safe side I'll remove fields that are not in the OrderSummary entity. We should start spiking about HSOI mapping. We may find many non supported fields.

File: commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                order.BillingCity = billingAddress.City;
+                order.BillingState = billingAddress.State;
+                order.BillingPostalCode = billingAddress.PostalCode;
+                order.BillingCountry = billingAddress.Country;
+                order.BillingLatitude = billingAddress.Latitude;
+                order.BillingLongitude = billingAddress.Longitude;
+                order.BillingEmailAddress = cart.getGuestEmailAddress();
+                order.BillingPhoneNumber = cart.getGuestPhoneNumber();
+                order.TaxLocaleType = cart.getTaxType().name();
+                order.EffectiveDate = Date.today();
```
Comment (szelenin-salesforce) [inline commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls:62]: @yan-huang-sf this field is missing in the OrderSummary but it's required for Order. Seems we'll have to deal with it somehow.

**Review by alexey-morozov-salesforce — APPROVED** [review]

File: commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls (hunk 4 above)
Comment (yan-huang-sf) [reply commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls:60]: Not sure how HSOI will set this field when they create order

**Review by yan-huang-sf — APPROVED** [review]

**Review by snilakandan13 — APPROVED** [review]

File: commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleIntegrationTest.cls [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        CartExtension.Cart cart = arrangeCart(new Decimal[]{1.5, 2.6});
+        CreateOrderSample createOrder = new CreateOrderSample();
+        // Act
+        Test.startTest();
+        CartExtension.CreateOrderResponse response = createOrder.createOrder(new CartExtension.CreateOrderRequest(cart, '123'));
+        Test.stopTest();
+        // Assert
+        Order order = response.getOrderGraph().getOrder();
+        Assert.areEqual('Total rounded up cost of products: ' + (2.0 + 3.0), order.Description);
+    }
```
Comment (jbisaSF) [inline commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleIntegrationTest.cls:31]: @szelenin-salesforce do we have >75% code coverage of `CreateOrderSample` with these tests?

Comment (szelenin-salesforce) [reply commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleIntegrationTest.cls:31]: @jbisaSF For Unit test it has 85%. See image below
image

For the integration test it has 100% coverage:
image

**Pull request merged**